addons/road-generator/nodes/road_lane_agent.gd

```diff
@@ -177,71 +177,111 @@ func test_move_along_lane(move_distance: float) -> Vector3:
+	var distance_left = 0
+	while check_next_offset > lane_length: # Target point is past the end of this curve
+		var lane_dir = _update_lane.lane_next if dir > 0 else _update_lane.lane_prior
+		var check_lane = _update_lane.get_node_or_null( lane_dir )
+		if ! is_instance_valid(check_lane):
+			distance_left = (check_next_offset - lane_length) * dir
+			check_next_offset = lane_length
+			break
+		check_next_offset -= lane_length
+		_update_lane = check_lane
+		lane_length = _update_lane.curve.get_baked_length()
+	while check_next_offset < 0:
+		var lane_dir = _update_lane.lane_next if dir < 0 else _update_lane.lane_prior
+		var check_lane = _update_lane.get_node_or_null( lane_dir )
+		if ! is_instance_valid(check_lane):
+			distance_left = check_next_offset * dir - init_offset
+			check_next_offset = 0
+			break
+		init_offset = 0
+		_update_lane = check_lane
+		check_next_offset += _update_lane.curve.get_baked_length()
+	if update_lane && _update_lane != current_lane:
 		assign_lane(_update_lane)
-
-	else: # Target point lies within the length of this curve
-		var ref_local = current_lane.curve.sample_baked(check_next_offset)
-		new_point = current_lane.to_global(ref_local)
-
+	var ref_local = _update_lane.curve.sample_baked(check_next_offset)
+	var new_point: Vector3 = _update_lane.to_global(ref_local)
+	if distance_left != 0: #workaround for missing connections
+		_update_lane = find_nearest_lane(pos - actor.global_transform.basis.z * dir)
+		if _update_lane != current_lane:
+			assign_lane(_update_lane)
 	return new_point
 
 
-## Input of -1 or 1 to assign left or right lane accordingly
+## Input of < 0 or > 0 to move abs(direction) amount of left or right lanes accordingly
 func change_lane(direction: int) -> int:
-	var _new_lane_path
-	if direction == 1:
-		_new_lane_path = current_lane.lane_right
-	elif direction == -1:
-		_new_lane_path = current_lane.lane_left
-	if not _new_lane_path:
-		# push_error("No lane to change to in target direction")
-		return FAILED
-	var _new_lane = current_lane.get_node_or_null(_new_lane_path)
-	if not is_instance_valid(_new_lane):
-		push_error("Invalid target lane change nodepath")
-		return FAILED
-	elif not _new_lane is RoadLane:
-		push_error("Target to change lane to is not a RoadLane")
-		return FAILED
-
+	if !direction:
+		return OK
+	var _new_lane = current_lane
+	var dec = sign(direction)
+	while direction != 0:
+		var _new_lane_path = _new_lane.lane_right if direction > 0 else _new_lane.lane_left
+		if not _new_lane_path:
+			# push_error("No lane to change to in target direction")
+			return FAILED
+		_new_lane = _new_lane.get_node_or_null(_new_lane_path)
+		if not is_instance_valid(_new_lane):
+			push_error("Invalid target lane change nodepath")
+			return FAILED
+		elif not _new_lane is RoadLane:
+			push_error("Target to change lane to is not a RoadLane")
+			return FAILED
+		direction -= dec
 	assign_lane(_new_lane)
 	return OK
 
+var close_time = 0.05
```

Comment:
Nitpick: Would prefer constants to be at the top of the file and assigned as const's unless actually changing